src/nfdump/exporter.c

```
@@ -174,14 +162,11 @@ int AddExporterInfo(exporter_info_record_t *exporter_record) {
                 while (next_free_slot_idx < MAX_EXPORTERS && exporter_list[next_free_slot_idx]) ++next_free_slot_idx;
 
                 // Did we find a new slot, or just reach the end of the list?
-                if (next_free_slot_idx >= MAX_EXPORTERS)
-                {
+                if (next_free_slot_idx >= MAX_EXPORTERS) {
```

Here you need to return a failure. `next_free_exporter_slot` is still >= MAX_EXPORTERS and if you only update `number_of_exporters_tracked` you overwrite memory in line 173.